### PR TITLE
Fix/0021375/5 3/composer readme unclear

### PR DESCRIPTION
**Pull request by d3r1w**
This PR addresses the ticket 0021375.

The following adjustments are made:
- Updated the composer README
- Removed git ignores because the production dependencies have to be committed and the dev dependencies are only in the composer.json / composer.lock. Therefore, it is no required to ignore something at all.
- Removed old components folder because it is likely a leftover of the robloach/component-installer which used require js to load these dependencies. (This is only implied by the names of the folder and the entries found in the git ignore file)
- Fixed indentation spaces -> tab

### CODE REVIEW

File: libs/composer/README.md [hunk 1]
```diff
@@ -1,11 +1,22 @@
...
+production and dev dependencies.
+
+The production dependencies are saved in the *require* section of the composer.json
+configuration. These are used by the production code of ILIAS. Tests and other development
+logic don't use this libraries directly.
+
+The dev dependencies are saved in the *require-dev* section of the compser.json configuration.
+These are libraries and tools dedicated to aid the development process. For example
+PHPUnit to run and create unit tests.
+
```
Comment (alex40724) [inline libs/composer/README.md:16]: Good!

**Review by alex40724 — CHANGES_REQUESTED** [review]
I think its much better. Thank you. I still would rephrase the sentence (Dev) "Ignore all directories which are added by installation (uncomment existing)" into something like "Add all directories that are newly added by the installation also to the .gitignore file".

**Comment by chfsx** [comment]
@alex40724 thanks for the review. I think we do no longer need the "ignore dev-dependencies" and  I changed the README accordingly.

File: libs/composer/README.md [hunk 2]
```diff
@@ -19,26 +30,67 @@
...
-updated using 
+Whenever a new ILIAS-Class in /Services\/* or /Modules\/* is introduced, the classmap should be 
+updated with the following command. 
 ```bash
 $ composer dump-autoload
-```
\ No newline at end of file
+```
+
+# Create library patch
```
Comment (klees) [inline libs/composer/README.md:71]: I would propose to disencourage that behaviour somehow, e.g. by putting a "has to be confirmed by JF" or "... TB".

**Comment by Amstutz** [comment]
Thanks a lot! I am not that deep into composer. However, from what I understand, this looks good.

File: libs/composer/README.md [hunk 3]
```diff
@@ -1,11 +1,22 @@
...
+logic don't use this libraries directly. 
+
+The dev dependencies are saved in the *require-dev* section of the compser.json configuration.
+These are libraries and tools dedicated to aid the development process. For example
+PHPUnit to run and create unit tests.
+
+### Dependencies for production
 - Add a new library using composer, e.g. "composer require filp/whoops"
-- Documents the usage and your wrapper class in composer.josn, e.g.:
+- Documents the usage and your wrapper class in composer.json, e.g.:
```
Comment (alex40724) [inline libs/composer/README.md]: Document(s) the usage...

Do we want to add the date of JF discussion to the JSON in this example now?

**Review by alex40724 — CHANGES_REQUESTED** [review]

File: libs/composer/README.md (hunk 3 above)
Comment (chfsx) [reply libs/composer/README.md]: yes I think so and added the lines to the readme and in our composer.json (without the effective dates since we will have to search a lot of JF-protocols or have one JF which accepts all of them

**Review by chfsx — APPROVED** [review]

**Comment by chfsx** [comment]
HI @alex40724 any other changes needed?

**Review by alex40724 — APPROVED** [review]

**Comment by alex40724** [comment]
No, thanks for the improvements!
